### PR TITLE
[Fix #6311] Prevent Style/Semicolon from breaking on single line if-then-else in assignment

### LINKED ISSUE #6311: An error occurred while Style/Semicolon cop
Running rubocop on the following code throws the following error. 

```
class Sample
  def limit
    limit = @company_limit if @by_company else @limit
    limit
  end
end
```

```
1 error occurred:
An error occurred while Style/Semicolon cop was inspecting rubocop_error.rb:3:4.
Errors are usually caused by RuboCop bugs.
Please, report your problems to RuboCop's issue tracker.
Mention the following information in the issue report:
0.51.0 (using Parser 2.5.1.2, running on ruby 2.4.2 x86_64-darwin16)
```

--------

## Expected behavior

No error message to be displayed

## Actual behavior

1 error occcurred

## Steps to reproduce the problem

Run rubocop in the code shared above. The error disappears when we remove the line `limit = @company_limit if @by_company else @limit` or rewrite in a different way.

## RuboCop version

Include the output of `rubocop -V` or `bundle exec rubocop -V` if using Bundler. Here's an example:

```
$ [bundle exec] rubocop -V
0.51.0 (using Parser 2.5.1.2, running on ruby 2.4.2 x86_64-darwin16)
```

### PR DESCRIPTION
**Pull request by Drenmi**
This cop was making the assumption that all multi-expression lines would have a semicolon on them, but it turns out that there are cases where this is not true, e.g.:

```ruby
def foo
  bar = baz if qux else quux
end
```

This change asserts that there is an actual semicolon before attempting to register an offense.

-----------------

Before submitting the PR make sure the following are checked:

* [X] Wrote [good commit messages][1].
* [X] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [X] Feature branch is up-to-date with `master` (if not - rebase it).
* [X] Squashed related commits together.
* [X] Added tests.
* [X] Added an entry to the [Changelog](../blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](../blob/master/CONTRIBUTING.md#changelog-entry-format).
* [X] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.
* [X] Run `rake default`. It executes all tests and RuboCop for itself, and generates the documentation.

[1]: http://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: spec/rubocop/cop/style/semicolon_spec.rb [hunk 1]
```diff
@@ -105,6 +105,16 @@ def foo(a) x(1); y(2); z(3); end
       RUBY
   end
 
+  context 'with a multi-expression line without a semicolon' do
```
Comment (bbatsov) [inline spec/rubocop/cop/style/semicolon_spec.rb]: Where's the multi-line expression here? :-)

**Comment by bbatsov** [comment]
Seems also that cop checks for two different problems and should be split at some point.

File: spec/rubocop/cop/style/semicolon_spec.rb (hunk 1 above)
Comment (Drenmi) [reply spec/rubocop/cop/style/semicolon_spec.rb]: It’s not a multi-line expression, it’s a multi-expression line ... 😅

Comment (bbatsov) [reply spec/rubocop/cop/style/semicolon_spec.rb]: Ops :-)

**Pull request merged**